e2e_test/test/e2e_test.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+void main() {
+  group('e2e test', () {
+    setUpAll(() {
+      expectSuccess(Process.runSync('pub', ['run', 'dazel', 'init']));
+    });
+
+    tearDown(() {
```

Comment:
It makes sense to clean between each test, we want to be able to validate the build output separately for each test and not end up catching previous build outputs.